sfge/src/main/java/com/salesforce/rules/ThreadableRuleExecutor.java

```diff
@@ -94,6 +94,7 @@ private static int submitRunners(
             completionService.submit(new CallableExecutor(submission, progressListener));
             submissionCount += 1;
         }
+        submissions.clear();
```

Comment:
Clearing the rule submission list immediately freed up a large part of heapspace since it didn't have to hold onto rule graphs on each submission object anymore.